File: app/src/main/java/com/yonder/exercise/ui/home/HomeFragmentPagerAdapter.java
package com.yonder.exercise.ui.home;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Created by YusufMac on 01/06/17.
 */


class HomeFragmentPagerAdapter extends FragmentPagerAdapter {
    private final List<Fragment> mFragmentList = new ArrayList<>();

    HomeFragmentPagerAdapter(FragmentManager manager) {
        super(manager);
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    void addFragments(Fragment... fragments) {
        mFragmentList.addAll(Arrays.asList(fragments));
    }

}

File: app/src/main/java/com/yonder/exercise/ui/home/HomeFragmentPagerAdapter.kt
package com.yonder.exercise.ui.home

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import java.util.ArrayList
import java.util.Arrays


 class HomeFragmentPagerAdapter(manager: FragmentManager) : FragmentPagerAdapter(manager) {
    private val mFragmentList = ArrayList<Fragment>()
    override fun getItem(position: Int): Fragment {
        return mFragmentList[position]
    }
    override fun getCount(): Int {
        return mFragmentList.size
    }
    fun addFragments(vararg fragments: Fragment) {
        mFragmentList.addAll(Arrays.asList(*fragments))
    }
}